### PR TITLE
Quark d2000 dts

### PR DESCRIPTION
**Pull request by SavinayDharmappa**
Add device tree support for mintvalley board and quark_d2000 microcontroller

### CODE REVIEW

File: dts/x86/quark_d2000_crb.dts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/dts-v1/;
+
+#include "intel_quark_d2000.dtsi"
+
+/ {
+	model = "intel,quark";
```
Comment (galak) [inline dts/x86/quark_d2000_crb.dts]: model should be the board model.

File: dts/x86/quark_d2000_crb.dts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	compatible = "intel,quark";
```
Comment (galak) [inline dts/x86/quark_d2000_crb.dts]: should include the board compat first (like we have in dts/x86/arduino_101.dts)

File: dts/x86/intel_quark_d2000.dtsi [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#include "skeleton.dtsi"
+#include "mem.h"
+
+/ {
+	cpus {
+		cpu@0 {
+			compatible = "intel,quark";
```
Comment (galak) [inline dts/x86/intel_quark_d2000.dtsi]: is the cpu core the same as on quark_se_c1000?  Is there really only one quark cpu core?

File: dts/x86/quark_d2000_crb.dts (hunk 2 above)
Comment (galak) [inline dts/x86/quark_d2000_crb.dts]: Actually this should be something like:

compatible = "intel,quark-d2000-crb", "intel-quark-d2000", "intel,quark";

(I'm not sure if the "intel,quark" actually makes sense).

**Review by galak — CHANGES_REQUESTED** [review]
Some fixups are needed.

File: dts/x86/intel_quark_d2000.dtsi (hunk 3 above)
Comment (SavinayDharmappa) [inline dts/x86/intel_quark_d2000.dtsi]: quark_d2000 and quark_se_c1000 have same cpu. But quark_se_c1000 also have sensor subsystem which is arc cpu which not present in quark_d2000

File: dts/x86/quark_d2000_crb.dts (hunk 2 above)
Comment (SavinayDharmappa) [inline dts/x86/quark_d2000_crb.dts]: will correct it

File: dts/x86/intel_quark_d2000.dtsi (hunk 3 above)
Comment (galak) [inline dts/x86/intel_quark_d2000.dtsi]: that's fine, just that the 'cpu' section here is meant for the type of core(s) in the SoC.  So from what you say, they should have the same compatible between quark_d2000 & quark_se_c1000.

File: dts/x86/quark_d2000_crb.dts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+/dts-v1/;
+
+#include "intel_quark_d2000.dtsi"
+
+/ {
+	model = "Mintvalley board";
```
Comment (galak) [inline dts/x86/quark_d2000_crb.dts]: seems like mintvalley is an internal name for the board, not seeing that referenced in our board docs for the board.  Probably not the best choice for model.

File: dts/x86/quark_d2000_crb.dts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	compatible = "intel,quark-d2000-crb","intel,quark-d2000";
```
Comment (galak) [inline dts/x86/quark_d2000_crb.dts]: nit, space after the comma

File: dts/x86/Makefile [hunk 6]
```diff
@@ -1,4 +1,5 @@
 ifeq ($(CONFIG_HAS_DTS),y)
 dtb-$(CONFIG_SOC_QUARK_SE_C1000)	=	arduino_101.dts_compiled
+dtb-$(CONFIG_SOC_SERIES_QUARK_D2000)	=	quark_d2000_crb.dts_compiled
```
Comment (galak) [inline dts/x86/Makefile]: should be in the board patch.

**Review by galak — CHANGES_REQUESTED** [review]
some minor things

**Review by galak — APPROVED** [review]

**Pull request merged**